nixos/modules/system/boot/luksroot.nix

```diff
@@ -444,12 +444,7 @@ let
                     # Try reading it from /dev/console with a timeout
                     IFS= read -t 1 -r pin
                     if [ -n "$pin" ]; then
-                       ${if luks.reusePassphrases then ''
-                         # Remember it for the next device
-                         echo -n "$pin" > /crypt-ramfs/passphrase
-                       '' else ''
-                         # Don't save it to ramfs. We are very paranoid
```

Comment:
Why do we always store to crypt-ramfs? I was just curious.

Reply:
I'm sure there are many other reasons I am not familiar with, but I do know that the `/crypt-ramfs/passphrase` path is used for reuse of a key to decrypt multiple drives (configured via `boot.initrd.luks.reusePassphrases`.

Reply:
Oh, so not writing to it even when reuse not being enabled could break behavior, you say?

Reply:
Actually, not it does not! This could use bash process substitution to read from an environment variable so that `--pin-source` will work. From the behavior I have seen, `--pin-source` requires the source to be a file. If process substitution is used, the pin doesn't have to be written to `/crypt-ramfs/...` and then conditionally removed.